### PR TITLE
Add marketplace, changelog & homepage links (if available)

### LINKED ISSUE #10: Link to the marketplace when "CHANGELOG.md not found".
It would be great if there was also an link to the marketplace when "_CHANGELOG.md not found_".

_Originally posted by @jerone in https://github.com/ryu1kn/vscode-extension-update-reporter/issues/2#issuecomment-585602560_

### PR DESCRIPTION
**Pull request by jerone**
Add links to:
1. MarketPlace.
2. Changelog (if file exists).
3. Homepage (if available in `package.json`).

Closes #10.

image

### CODE REVIEW

**Comment by jerone** [comment]
Link to changelog will follow soon.

**Comment by ryu1kn** [comment]
@jerone , thanks so much for the improvements here.

By the way, I actually like what @hansWurst-creator suggested at https://github.com/ryu1kn/vscode-extension-update-reporter/issues/10#issuecomment-717834577 as it's the same behaviour with the VS Code extension page. Also with a home icon, it may not be too obvious what icon means or what it does 🤔

**Comment by jerone** [comment]
I understand that suggestion, but I was thinking of implementing [collapsing sections](https://github.com/ryu1kn/vscode-extension-update-reporter/issues/11#issuecomment-937622577). This allows users to hide whole extension changelogs when we cannot correctly parse version ranges, and thus render the whole changelog. Making the header an link would leave less room for collapsing/expanding sections.

Let me see what I can do with collapsing sections...

**Comment by jerone** [comment]
Rebased.
Updated the screenshot.

**Comment by ryu1kn** [comment]
Thanks @jerone for this improvement!

I'm still feeling 🏠 emoji may not be the clearest expression for the user to find this capability. What do you think about making it a text like "go to its homepage"?

By the way, should we take the user to the homepage? Other options would be a marketplace or a repository. I wonder if marketplace as that location would most likely to be available.

**Comment by jerone** [comment]
Shall I add homepage, marketplace and link to changelog inside the collapsible content?
Just before the changelog versions.

Something like:
```md
# Extension A

[Homepage](https://github.com/ryu1kn/vscode-extension-update-reporter) ● [Marketplace](linkie) ● [Changelog](linkie) 
```
Where all links are only added when available.

**Comment by jerone** [comment]
Suggestion:
image

**Comment by jerone** [comment]
Added tests.
And made the changelog link optional.
You can see all three situations in this screenshot:
1. No homepage.
2. Has changelog & homepage.
3. No changelog.

image

File: src/test/lib/markdown-report.test.ts [hunk 1]
```diff
@@ -96,6 +102,8 @@ describe('Markdown Report', () => {
         ## EXT_NAME_3 \`EXT3\`
         </summary>
 
+        [Marketplace](https://marketplace.visualstudio.com/items/EXT3) ● [Changelog](https://marketplace.visualstudio.com/items/EXT3/changelog)
+
         Couldn't parse the changelog. [View it on Marketplace](https://marketplace.visualstudio.com/items/EXT3/changelog).
```
Comment (ryu1kn) [inline src/test/lib/markdown-report.test.ts]: Shall we change it to something like

_Please view it on Marketplace by following the Changelog link above_

**Review by ryu1kn — COMMENTED** [review]
Thank you @jerone !

File: src/test/lib/markdown-report.test.ts (hunk 1 above)
Comment (jerone) [reply src/test/lib/markdown-report.test.ts]: With or without the first sentence?

> Couldn't parse the changelog. Please view it on Marketplace by following the Changelog link above.

Comment (ryu1kn) [reply src/test/lib/markdown-report.test.ts]: With the first sentence 😉

**Comment by ryu1kn** [comment]
Thank you for the improvements! 🙏

**Comment by ryu1kn** [comment]
Published as v1.2.0, thank you @jerone ! 🎉

**Pull request merged**